plonk-core/src/proof_system/widget/lookup.rs

```diff
@@ -81,3 +209,78 @@ where
         lc(vec![w_l, w_r, w_o, w_4], zeta)
     }
 }
+
+/// Permutation Verifier Key
+#[derive(CanonicalDeserialize, CanonicalSerialize, derivative::Derivative)]
+#[derivative(
+    Clone(bound = ""),
+    Debug(bound = "PCC: std::fmt::Debug"),
+    Eq(bound = "PCC: Eq"),
+    PartialEq(bound = "PCC: PartialEq")
+)]
+pub struct VerifierKey<PCC>
+where
+    PCC: PCCommitment + Default,
+{
+    /// Lookup Selector Commitment
+    pub q_lookup: PCC,
+}
+
+impl<PCC> VerifierKey<PCC>
+where
+    PCC: PCCommitment + Default,
+{
+    /// Computes the linearisation commitments.
+    pub fn compute_linearisation_commitment<F: FftField>(
+        &self,
+        scalars: &mut Vec<F>,
+        points: &mut Vec<PCC>,
+        evaluations: &ProofEvaluations<F>,
+        (delta, epsilon, zeta): (F, F, F),
+        lookup_sep: F,
+        l1_eval: F,
+        z2_comm: PCC,
+        h1_comm: PCC,
+    ) {
+        let f = {
+            let f_0 = evaluations.wire_evals.a_eval
+                - evaluations.lookup_evals.f_eval
+                + zeta * evaluations.wire_evals.b_eval
+                + zeta * zeta * evaluations.wire_evals.c_eval
+                + zeta * zeta * zeta * evaluations.wire_evals.d_eval;
+            f_0 * lookup_sep
+        };
+
+        scalars.push(f);
+        points.push(self.q_lookup.clone());
+
+        // (1 + δ) * (ε + f_bar) * (ε(1+δ) + t_bar + δ*tω_bar) *  lookup_sep^2
```

Comment:
+ L_1 * lookup_sep^3

Reply:
line 267 has this computation

Reply:
but we should add it in the comment too!

Reply:
Can you explain this comment? I'm not sure what you are trying to point out.

Reply:
My apologies I just saw your reply to this, I think it was on a different commit.